docs/strategies/circuit-breaker.md

```diff
@@ -99,3 +99,229 @@ await manualControl.CloseAsync();
+    }
+}
+
+var sleepDurationProvider = GetSleepDuration().GetEnumerator();
+sleepDurationProvider.MoveNext();
+
+var circuitBreaker = new ResiliencePipelineBuilder()
+    .AddCircuitBreaker(new()
+    {
+        ShouldHandle = new PredicateBuilder().Handle<HttpRequestException>(),
+        BreakDuration = TimeSpan.FromSeconds(0.5),
+        OnOpened = async args =>
+        {
+            await Task.Delay(sleepDurationProvider.Current);
+            sleepDurationProvider.MoveNext();
+        }
+
+    })
+    .Build();
+```
+<!-- endSnippet -->
+
+**Reasoning**:
+- The lowest value of break duration is half second
+  - That means each sleep is actually takes `sleepDurationProvider.Current` + half second
+- You might think that setting the `BreakDuration` to the `sleepDurationProvider.Current` solves the problem
+  - But it doesn't since the `BreakDuration` is set only once and not re-evalutated at every break
+
+<!-- snippet: circuit-breaker-anti-pattern-2-ext -->
+```cs
+circuitBreaker = new ResiliencePipelineBuilder()
+    .AddCircuitBreaker(new()
+    {
+        ShouldHandle = new PredicateBuilder().Handle<HttpRequestException>(),
+        BreakDuration = sleepDurationProvider.Current,
+        OnOpened = async args =>
+        {
+            Console.WriteLine($"Break: {sleepDurationProvider.Current}");
+            sleepDurationProvider.MoveNext();
+        }
+
+    })
+    .Build();
+```
+<!-- endSnippet -->
+
+✅ DO
+
+The `CircuitBreakerStartegyOptions` currently do not provide a way to define break durations dynamically.
+This might be re-evaluted later until that use the first example as a workaround with caution.
+
+### 3 - Wrapping each endpoint with a circuit breaker
+
+Imagine that you have to call N number of services via `HttpClient`s.
+You want to decorate all downstream calls with the service-aware Circuit Breaker.
+
+❌ DON'T
+
+Use a collection of Circuit Breakers and explicitly call `ExecuteAsync`
+
+<!-- snippet: circuit-breaker-anti-pattern-3 -->
+```cs
+// Defined in a common place
+var uriToCbMappings = new Dictionary<Uri, ResiliencePipeline>
+{
+    { new Uri("https://downstream1.com"), GetCircuitBreaker() },
+    // ...
+    { new Uri("https://downstreamN.com"), GetCircuitBreaker() }
+};
+
+// Used in the downstream 1 client
+var downstream1Uri = new Uri("https://downstream1.com");
+await uriToCbMappings[downstream1Uri].ExecuteAsync(CallXYZOnDownstream1, CancellationToken.None);
+```
+<!-- endSnippet -->
+
+**Reasoning**:
+- In every places where you use an `HttpClient` you have to have a reference to the  `uriToCbMappings`
+- It is your responsibility to decorate each and every network call with the related circuit breaker
+
+✅ DO
+
+Use named and typed `HttpClient`s
+
+```cs
+foreach (string uri in uris)
+{
+    builder.Services
+      .AddHttpClient<IResilientClient, ResilientClient>(uri, client => client.BaseAddress = new Uri(uri))
+      .AddPolicyHandler(GetCircuitBreaker().AsAsyncPolicy<HttpResponseMessage>());
```

Comment:
Later, we will use `AddResilienceHandler` from the new `Microsoft.Extensions.Http.Resilience` package which is the successor to the other one.